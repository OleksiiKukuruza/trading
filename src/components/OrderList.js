import React from "react";
import styled from "styled-components";
import { Cell, HeaderRow, Row } from "./Table";

const StyledList = styled.div`
  margin: ${props => 2 * props.theme.spacingUnit}px;
`;

const StyledListContent = styled.div`
  border: 1px solid ${props => props.theme.background};
`;

const OrderList = ({ orders, title }) => (
  <StyledList>
    {title}
    <StyledListContent>
      <HeaderRow>
        <Cell>Order Id</Cell>
        <Cell>Price</Cell>
        <Cell>Quantity</Cell>
      </HeaderRow>
      {orders.map(order => (
        <Row key={order.id}>
          <Cell>{order.id}</Cell>
          <Cell>{order.price}</Cell>
          <Cell>{order.quantity}</Cell>
        </Row>
      ))}
    </StyledListContent>
  </StyledList>
);

export default OrderList;
