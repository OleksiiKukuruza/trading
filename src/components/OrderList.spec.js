import React from 'react';
import { shallow } from 'enzyme';
import OrderList from './OrderList';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('OrderList', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup(OrderList, {
      orders: [
        { id: 1, price: 99, quantity: 10, type: 'sell' },
        { id: 3, price: 101, quantity: 10, type: 'sell' }
      ],
      title: 'Title'
    });
    expect(wrapper).toMatchSnapshot();
  });
});
