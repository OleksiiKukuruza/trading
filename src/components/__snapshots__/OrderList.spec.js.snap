// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList renders without crashing 1`] = `
<div>
  Title
  <styled.div>
    <Styled(styled.div)>
      <styled.div>
        Order Id
      </styled.div>
      <styled.div>
        Price
      </styled.div>
      <styled.div>
        Quantity
      </styled.div>
    </Styled(styled.div)>
    <styled.div
      key="1"
    >
      <styled.div>
        1
      </styled.div>
      <styled.div>
        99
      </styled.div>
      <styled.div>
        10
      </styled.div>
    </styled.div>
    <styled.div
      key="3"
    >
      <styled.div>
        3
      </styled.div>
      <styled.div>
        101
      </styled.div>
      <styled.div>
        10
      </styled.div>
    </styled.div>
  </styled.div>
</div>
`;
